src/script/common/c_converter.cpp

```diff
@@ -51,23 +51,32 @@ if (value < F1000_MIN || value > F1000_MAX) { \
 #define CHECK_POS_TAB(index) CHECK_TYPE(index, "position", LUA_TTABLE)
 
 
+void push_float_string(lua_State *L, float value)
+{
+	std::stringstream ss;
+	std::string str;
+	ss << value;
+	str = ss.str();
+	lua_pushstring(L, str.c_str());
+}
+
 void push_v3f(lua_State *L, v3f p)
 {
 	lua_newtable(L);
-	lua_pushnumber(L, p.X);
+	push_float_string(L, p.X);
 	lua_setfield(L, -2, "x");
-	lua_pushnumber(L, p.Y);
+	push_float_string(L, p.Y);
 	lua_setfield(L, -2, "y");
-	lua_pushnumber(L, p.Z);
+	push_float_string(L, p.Z);
 	lua_setfield(L, -2, "z");
 }
 
 void push_v2f(lua_State *L, v2f p)
 {
 	lua_newtable(L);
-	lua_pushnumber(L, p.X);
+	push_float_string(L, p.X);
 	lua_setfield(L, -2, "x");
-	lua_pushnumber(L, p.Y);
+	push_float_string(L, p.Y);
```

Comment:
Please do not change these functions, as several mods could stop working properly with different value types. See `l_get_animation`, `l_hud_get`, `push_object_properties` and others.
`push_v2_float_string()` and `push_v3_float_string()` would be more descriptive anyway.

Reply:
What if I change it to `double` instead of just `std::string`? Will it impact performance a lot? Or just create new function?